lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -421,7 +421,7 @@ private Set<Location> listGen2Directories(AzureLocation location)
             pathItems = fileSystemClient.listPaths();
```

Comment:
I tried adding one, but it also passes before adding `urlEncode` calls. This is because the tests also create files using the filesystem module. Renaming files wasn't working, that's why I added a test case for it.